soc/arm/Kconfig

```diff
@@ -57,6 +57,30 @@ config NRF_SPU_RAM_REGION_SIZE
 	  RAM region size for the NRF_SPU peripheral
 endif
 
+if HAS_HW_NRF_MPU
+config NRF_MPU_FLASH_REGION_SIZE
```

Comment:
This one is strictly an nRF51-specific feature so I would rather see this under the nrf51 soc Kconfig